### PR TITLE
Use Xsource:3

### PR DESCRIPTION
**Pull request by saeltz**
Scala 3 is deprecating package objects with inheritance (see https://github.com/scala/scala-dev/issues/441). We were using them quite a lot.
This uses the compiler flag `-Xsource:3` to err on features deprecated in Scala 3 and changes package objects to normal objects which requires some shifting of files to keep the imports stable.
This is the first PR of many to support Scala 3.

### CODE REVIEW

File: src/main/scala/scynamo/syntax/AttributeValueDslOps.scala [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+import java.util
+
+object attributevalue extends AttributeValueDsl
+
 trait AttributeValueDsl {
   implicit def toAttributeValueOps(attributeValue: AttributeValue): AttributeValueDslOps = new AttributeValueDslOps(attributeValue)
 }
 
-class AttributeValueDslOps(val attributeValue: AttributeValue) extends AnyVal {
+class AttributeValueDslOps(val attributeValue: AttributeValue) extends AttributeValueDsl {
```
Comment (joroKr21) [inline src/main/scala/scynamo/syntax/AttributeValueDslOps.scala:18]: Why this change?

Comment (saeltz) [reply src/main/scala/scynamo/syntax/AttributeValueDslOps.scala:18]: Because `asEither` in line 45 below uses the `asOption` and needs the implicit conversion.

Comment (joroKr21) [reply src/main/scala/scynamo/syntax/AttributeValueDslOps.scala:18]: Yeah but can't we just import it there? This is really weird and less efficient.

Comment (saeltz) [reply src/main/scala/scynamo/syntax/AttributeValueDslOps.scala]: Like in https://github.com/moia-oss/scynamo/pull/232/commits/bc12161bfef9d173cd6097fd369111f210dcf904?

**Review by joroKr21 — APPROVED** [review]
Yes 👍
